userspace/libsinsp/threadinfo.cpp

```diff
@@ -39,13 +39,29 @@ static void copy_ipv6_address(uint32_t* dest, uint32_t* src) {
 }
 
 sinsp_threadinfo::sinsp_threadinfo(
+        const sinsp_mode& mode,
+        const sinsp_network_interfaces& network_interfaces,
```

Comment:
It looks like a lot of the information added to the threadinfo here is only used in fix_sockets_coming_from_proc, which is only called in sinsp::init(). If that's the case, maybe it would be better to either pass that information explicitly as arguments to fix_sockets_coming_from_proc, or perhaps pass the entire inspector as an argument to fix_sockets_coming_from_proc.

Adding so many references to each threadinfo will increase memory usage when there are a lot of threads, too.

Reply:
Mh... I'm not understanding which `fix_sockets_coming_from_proc` method are we talking about, I cannot see where these dependencies we are injecting are used both in `sinsp_thread_manager::fix_sockets_coming_from_proc()` and `sinsp_threadinfo::fix_sockets_coming_from_proc()`, apart from `m_bound_server_ports`.
Anyway, I agree, we are passing a lot of information to the thread info in this way, and this will increase memory consumption a lot if the number of sinsp_threadinfo is huge. I would suggest to pack all these parameters in a structure that we are going to pass to each sinsp_threadinfo at construction time from sinsp: something like `sinsp_threadinfo_params`. WDYT?

Reply:
I mean that the only place where `sinsp_threadinfo::fix_sockets_coming_from_proc` is called is from `sinsp_thread_manager::fix_sockets_coming_from_proc`, and the only place where that is called is from https://github.com/falcosecurity/libs/blob/master/userspace/libsinsp/sinsp.cpp#L343.

So instead of adding all the information from the inspector to each threadinfo, you could just pass the information as arguments to `sinsp_thread_manager::fix_sockets_coming_from_proc` and then `sinsp_threadinfo::fix_sockets_coming_from_proc`, so you don't have to save it at all.

@gnosek also covered this, so I think you already addressed it by switching params to a single struct.